test/select_io_dset.c

```diff
@@ -782,6 +820,665 @@ test_cmpd_with_bkg(hid_t fid, unsigned chunked)
+        if (H5Sselect_none(file_sids[i]) < 0)
+            TEST_ERROR;
+    }
+
+    if (H5Dwrite_multi(ndsets, dset_dids, mem_tids, mem_sids, file_sids, H5P_DEFAULT, wbufs) < 0)
+        TEST_ERROR;
+
+    if (H5Dread_multi(ndsets, dset_dids, mem_tids, mem_sids, file_sids, H5P_DEFAULT, rbufs) < 0)
+        TEST_ERROR;
+
+    /* Verify data read */
+    for (i = 0; i < (int)ndsets; i++)
+        for (j = 0; j < DSET_SELECT_DIM; j++)
+            if (i == 0) { /* dset0 */
+                if (wbufi[i][j].a != rbufi[i][j].a || ((4 * (int)j) + 1) != rbufi[i][j].b ||
+                    wbufi[i][j].c != rbufi[i][j].c || ((4 * (int)j) + 3) != rbufi[i][j].d) {
+                    H5_FAILED();
+                    HDprintf("    Read different values than written.\n");
+                    HDprintf("    For dset %d at index %d\n", i, j);
+                    TEST_ERROR;
+                }
+            }
+            else { /* other datasets */
+                for (j = 0; j < DSET_SELECT_DIM; j++)
+                    if ((4 * (int)j) != rbufi[i][j].a || ((4 * (int)j) + 1) != rbufi[i][j].b ||
+                        ((4 * (int)j) + 2) != rbufi[i][j].c || ((4 * (int)j) + 3) != rbufi[i][j].d) {
+                        H5_FAILED();
+                        HDprintf("    Read different values than written.\n");
+                        HDprintf("    For dset %d at index %d\n", i, j);
+                        TEST_ERROR;
+                    }
+            }
+
+    /* Case c */
+    mm = HDrandom() % (int)ndsets;
+    if (!mm)
+        mm++;
+
+    /* Update data in rbufi for dset1 with new unique values */
+    for (j = 0; j < DSET_SELECT_DIM; j++) {
+        rbufi[mm][j].a = (4 * j) + (2 * DSET_SELECT_DIM);
+        rbufi[mm][j].b = (4 * j) + (2 * DSET_SELECT_DIM) + 1;
+        rbufi[mm][j].c = (4 * j) + (2 * DSET_SELECT_DIM) + 2;
+        rbufi[mm][j].d = (4 * j) + (2 * DSET_SELECT_DIM) + 3;
+    }
+
+    /* Create a compound type same size as s1_t */
+    if ((ss_bc_tid = H5Tcreate(H5T_COMPOUND, sizeof(s1_t))) < 0)
+        FAIL_STACK_ERROR;
+
+    /* but contains only subset members of s1_t */
+    if (H5Tinsert(ss_bc_tid, "b", HOFFSET(s1_t, b), H5T_NATIVE_INT) < 0 ||
+        H5Tinsert(ss_bc_tid, "c", HOFFSET(s1_t, c), H5T_NATIVE_INT) < 0)
+        FAIL_STACK_ERROR;
+
+    /* Datatype setting for read from dset1 */
+    mem_tids[mm] = ss_bc_tid;
+
+    /* Reset memory and file space for <mm> dataset */
+    if (H5Sselect_all(mem_sids[mm]) < 0)
+        FAIL_STACK_ERROR;
+    if (H5Sselect_all(file_sids[mm]) < 0)
+        FAIL_STACK_ERROR;
+
+    /* Untouched memory and file space for other datasets */
+    for (i = 0; i < (int)ndsets; i++) {
+        if (i == 0 || i == mm)
+            continue;
+        if (H5Sselect_none(mem_sids[i]) < 0)
+            TEST_ERROR;
+        if (H5Sselect_none(file_sids[i]) < 0)
+            TEST_ERROR;
+    }
+
+    /* Datatype setting for read from <mm> dataset */
+    mem_tids[mm] = ss_bc_tid;
```

Comment:
This line seems to be repeated unnecessarily

Reply:
Removed.